### ISSUE TITLE
Using queue-groups when subscribing to $SYS.ACCOUNT.> across routes

### ISSUE THREAD

**Issue by n-holmstedt** [nats-io/nats-server#5187]
### Observed behavior

I am currently using two superclusters of nats, one in test and one in production.

When subscribing to $SYS.ACCOUNT.> without any queue i can receive account CONNECT/DISCONNECT events from all clusters across routes. 

When subscribing with a queue group i can only receive events from the cluster i am currently connected to. 

Very simillar to https://github.com/nats-io/nats-server/issues/3177

### Expected behavior

Client within a queue group should be able to receive account events across a super cluster.

### Server and client version

nats-server: v2.10.11 (Both test & production)

~ # nats --version
v0.1.1

### Host environment

Official docker image in kubernetes using official helm charts. 

### Steps to reproduce

``nats -s <redacted>--creds=<SYS_PATH>/sys.creds sub '$SYS.ACCOUNT.*.*'``

``nats -s <redacted>--creds=<SYS_PATH>/sys.creds sub '$SYS.ACCOUNT.*.*' --queue=foo``

This behaviour has been verfied both in test and production.

**Comment by n-holmstedt** [comment]
Sorry if i was unclear. I am aware that if you have multiple clients in multiple clusters messages will be routed to the nearest clients.

Above behavior happens even if you have only one single instance in a unique queue-group. This single instance will only be able to receive CONNECT/DISCONNECT events from the local cluster.

**Comment by ripienaar** [comment]
Right, sorry yeah I can confirm this is a current problem on latest 2.11 nightly.

/cc @nats-io/server

**Comment by ripienaar** [comment]
```golang
// https://github.com/nats-io/nats-server/issues/5187
func TestCrossGatewayConnectEvents(t *testing.T) {
	ca := createClusterEx(t, true, 5*time.Millisecond, true, "A", 2)
	defer shutdownCluster(ca)
	cb := createClusterEx(t, true, 5*time.Millisecond, true, "B", 2, ca)
	defer shutdownCluster(cb)

	sysA, err := nats.Connect(ca.randomServer().ClientURL(), nats.UserInfo("sys", "pass"))
	checkError(err, nil, t)
	defer sysA.Close()

	checkEvents := func(t *testing.T, name string, queue bool) {
		t.Helper()

		t.Run(name, func(t *testing.T) {
			sysA, err := nats.Connect(ca.randomServer().ClientURL(), nats.UserInfo("sys", "pass"))
			checkError(err, nil, t)
			defer sysA.Close()

			var sub1 *nats.Subscription
			if queue {
				sub1, err = sysA.QueueSubscribeSync("$SYS.ACCOUNT.*.CONNECT", "x")
			} else {
				sub1, err = sysA.SubscribeSync("$SYS.ACCOUNT.*.CONNECT")
			}
			checkError(err, nil, t)

			cA, err := nats.Connect(ca.randomServer().ClientURL(), nats.UserInfo("foo", "pass"))
			checkError(err, nil, t)
			defer cA.Close()

			msg, err := sub1.NextMsg(time.Second)
			checkError(err, nil, t)
			if msg.Subject != "$SYS.ACCOUNT.FOO.CONNECT" {
				t.Fatalf("expected subject $SYS.ACCOUNT.FOO.CONNECT got %v", msg.Subject)
			}

			cB, err := nats.Connect(cb.randomServer().ClientURL(), nats.UserInfo("foo", "pass"))
			checkError(err, nil, t)
			defer cB.Close()

			msg, err = sub1.NextMsg(time.Second)
			checkError(err, nil, t)
			if msg.Subject != "$SYS.ACCOUNT.FOO.CONNECT" {
				t.Fatalf("expected subject $SYS.ACCOUNT.FOO.CONNECT got %v", msg.Subject)
			}
		})
	}

	checkEvents(t, "Unqueued", false)
	checkEvents(t, "Queued", true)
}
```

**Comment by neilalexander** [comment]
This will be fixed in 2.10.12, thanks for the report!
